### PR TITLE
1222 chainspec documentation

### LINKED ISSUE #1222: Document the chainspec.toml in docs.casper.network
- Double check the requirements
- Bring the information from the inline [chainspec.toml ](https://github.com/casper-network/casper-node/blob/release-1.5.1/resources/production/chainspec.toml) into the docs
- Double check that we want this in the feat-1.5.0 branch (although the build is 1.5.2)

### PR DESCRIPTION
**Pull request by ipopescu**
### What does this PR fix/introduce?

This PR adds the content from the 1.5.2 Mainnet chainspec to our documentation, with some small additions introduced by commit 969cf9ef94c063edf887705131d48a2dc711585e. The main file is [here](https://github.com/casper-network/docs/blob/969cf9ef94c063edf887705131d48a2dc711585e/source/docs/casper/operators/setup-network/chain-spec.md).

I left a few TODOs for open questions in the file so I can resolve them during the review process:

1. For the 'hard_reset' description, can we say: "Specifies whether or not the latest blocks will be cleared back to the switch block just before the activation point in case of resetting the node"? When I see 'hard_reset', I think this happens when a node needs to be reset.
2. Where can I find the genesis_hash to link to it? It is mentioned in the `name` attribute under #network.
3. When prune_batch_size=1 means it is ON? Do we support it?
4. Can I assume that when a boolean is set to 1 (ON), we also support the 0 (OFF) option? So I can add the possibilities in the description.
5. I see a section with settings for Highway. Are there similar settings for Zug? If so, could I get information on them so we can add them to the documentation?
6. Could you please review the additions I made with this commit 969cf9ef94c063edf887705131d48a2dc711585e, or the entire file if it is easier? I want to make sure that my additions are technically correct.

Closes #1222 

### Checklist

- [x] Docs are successfully building - `yarn install && yarn run build`.
- [x] For new **internal** links I used *relative file paths* (with .md extension) - e.g. `../../faq/faq-general.md` - instead of introducing *absolute file path*, or *relative/absolute URL*.
- [x] All external links have been verified with `yarn run check:externals`.
- [x] My changes follow the [Casper docs style guidelines](https://docs.casper.network/resources/contribute-to-docs/).

### Reviewers
@sacherjj and @ACStoneCL

### CODE REVIEW

**Comment by sacherjj** [comment]
1 - We have not come up with a reason for NOT having hard_reset on.  It does pop off blocks and deploys back to the end of the last era before the activation point.  So if for some reason the majority of weight did not stage upgrade and then does, it would unwrap the chain and go back. 

This is a hard_reset at the upgrade point.  So "resetting the node" isn't accurate as much as resetting the network progress.

**Comment by sacherjj** [comment]
2 - I believe this is talking about the state_root_hash of the genesis block.

**Comment by sacherjj** [comment]
3 - prune_batch_size was used for tip pruning in 1.4.15. This was the number of eras to process per block during this process. I believe this will be used when we enable state pruning to remove a bit of old data with each block, but not sure about this.

**Comment by sacherjj** [comment]
5 - core consensus_protocol enables which consensus protocol is used.  We have run tests that switch between zug and highway with an upgrade every era (for chainpec.toml change)  I don't think we will have the chainspec.toml for zug finalized until 2.0, so not worth pre-lim docs that might change.

**Comment by ipopescu** [comment]
> This is a hard_reset at the upgrade point. So "resetting the node" isn't accurate as much as resetting the network progress.
Updated to the following with commit ca0ee80ab9bdc603afde97ac1838a5d8b2946bea:
When set to true, clear blocks and deploys back to the switch block (the end of the last era) just before the activation point. Used during the upgrade process to reset the network progress. In most cases, this setting should be true.

**Comment by ipopescu** [comment]
> 3/4 - prune_batch_size was used for tip pruning in 1.4.15...

Updated to the following for more context, with commit ca0ee80ab9bdc603afde97ac1838a5d8b2946bea:

```
Global state prune batch size for tip pruning in version 1.4.15. Possible values:
- 0 when the feature is OFF
- Integer if the feature is ON, representing the number of eras to process per block.
```

**Comment by ipopescu** [comment]
@sacherjj @ACStoneCL this is ready for review now. I've made updates based on Joe's feedback. Thanks, Joe!

File: source/docs/casper/operators/setup-network/chain-spec.md [hunk 1]
```diff
@@ -2,23 +2,259 @@
...
+|session_args_max_length | The limit of length of serialized session code arguments. | 1024|
+|native_transfer_minimum_motes | The minimum amount in motes for a valid native transfer. | 2_500_000_000|
+
+## wasm
+
+The following are Wasm-related settings.
+
+|Attribute         |Description                                    | Mainnet Setting |
+|----------------- |-----------------------------------------------|-----------------|
+|max_memory | Amount of free memory (in 64kB pages) each contract can use for its stack. | 64|
```
Comment (ACStoneCL) [inline source/docs/casper/operators/setup-network/chain-spec.md:90]: It's written as 64kB here and 64kb on line 121.

File: source/docs/casper/operators/setup-network/genesis.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+title: Genesis
+---
+
+# The Genesis Block {#the-genesis-block}
+
+The Casper node software creates a genesis block from the following input files:
+
+- chainspec.toml
```
Comment (ACStoneCL) [inline source/docs/casper/operators/setup-network/genesis.md:9]: I wonder if we should italicize the .toml files - I think that's what we've historically done, but I'm not overly concerned about it.

**Review by ACStoneCL — APPROVED** [review]
Content looks great to me, just a few suggestions.

File: source/docs/casper/operators/setup-network/genesis.md (hunk 2 above)
Comment (ipopescu) [reply source/docs/casper/operators/setup-network/genesis.md]: Thanks for catching this! I used backticks to match the formatting of other file names, including "casper-node/resources/production/chainspec.toml". Updated with commit f12c60f3cb2185786f5067f504bd513faab14f47.

File: source/docs/casper/operators/setup-network/chain-spec.md (hunk 1 above)
Comment (ipopescu) [reply source/docs/casper/operators/setup-network/chain-spec.md]: I looked up the suggested spelling and added a space: 64 kB. Updated with f12c60f3cb2185786f5067f504bd513faab14f47.

File: source/docs/casper/operators/setup-network/chain-spec.md [hunk 3]
```diff
@@ -2,23 +2,259 @@
...
+|maximum_round_length | Highway dynamically chooses its round length between `minimum_block_time` and `maximum_round_length`. | '132seconds'|
+|reduced_reward_multiplier | The factor by which rewards for a round are multiplied if the greatest summit has ≤50% quorum, i.e., no finality. Expressed as a fraction (1/5 by default on Mainnet). | [1, 5]|
+
+## deploys
+
+These settings manage deploys and their lifecycle.
+
+|Attribute         |Description                                    | Mainnet Setting |
+|----------------- |-----------------------------------------------|-----------------|
+|max_payment_cost | The maximum number of Motes allowed to be spent during payment. 0 means unlimited. | '0'|
```
[the remark this answers is not in the document]
Comment (ipopescu) [reply source/docs/casper/operators/setup-network/chain-spec.md]: Updated with commit f12c60f3cb2185786f5067f504bd513faab14f47.

File: source/docs/casper/operators/setup-network/chain-spec.md [hunk 4]
```diff
@@ -2,23 +2,259 @@
 title: The Chainspec
 ---
 
-# The Chain Specification {#the-chain-specification}
+# The Blockchain Specification {#the-chain-specification}
 
-The Casper node software creates a genesis block from the following input files:
+The blockchain specification, or `chainspec`, is a collection of configuration settings describing the network state at genesis and upgrades to basic system functionality (including system contracts and gas costs) occurring after genesis. This page describes each field in the chainspec, based on the [version 1.5.2](https://github.com/casper-network/casper-node/blob/release-1.5.2/resources/production/chainspec.toml) of the Casper node. The chainspec can and should be customized for private networks. The chainspec attributes are divided into categories based on what they are configuring.
```
[the remark this answers is not in the document]
Comment (ipopescu) [reply source/docs/casper/operators/setup-network/chain-spec.md]: Updated with commit f12c60f3cb2185786f5067f504bd513faab14f47.

**Comment by ipopescu** [comment]
Thanks, @ACStoneCL!

**Review by sacherjj — APPROVED** [review]
Looks good.

**Pull request merged**